pydantic/type_adapter.py

```diff
@@ -118,6 +118,12 @@ class TypeAdapter(Generic[T]):
 
     **Note:** `TypeAdapter` instances are not types, and cannot be used as type annotations for fields.
 
+    **Note:** By default, `TypeAdapter` does not respect the
+    [`defer_build=True`][pydantic.config.ConfigDict.defer_build] setting in the
+    [`model_config`][pydantic.BaseModel.model_config] or in the `TypeAdapter` constructor `config`. You need to also
+    explicitly set [`defer_build_mode="always"`][pydantic.config.ConfigDict.defer_build_mode] of the config to defer
+    the model validator and serializer construction. This is required due to backwards compatibility reasons.
```

Comment:
Maybe something a bit more positive like: `Thus, this feature is opt-in to ensure backwards compatibility`